tests/core/tools.js

```diff
@@ -773,6 +789,32 @@
 
 			// Check standard selector.
 			assert.areSame( escapedSelector, 'aaa', 'standard selector' );
+		},
+
+		// #810
+		'test getMouseButton': function() {
+			if ( CKEDITOR.env.ie && CKEDITOR.env.version < 9 ) {
```

Comment:
Let's simplify this, rather than doing two tests for ie8 and non ie8, please add a variable `isIe8` and assert would look like:

```js
generateMouseButtonAsserts( [
	[ CKEDITOR.MOUSEBUTTON_LEFT, isIe8 ? 1 : CKEDITOR.MOUSEBUTTON_LEFT ],
	[ CKEDITOR.MOUSEBUTTON_MIDDLE, isIe8 ? 4 : CKEDITOR.MOUSEBUTTON_MIDDLE ],
	[ CKEDITOR.MOUSEBUTTON_RIGHT, isIe8 ? 2 : CKEDITOR.MOUSEBUTTON_RIGHT ]
] );
```

Reply:
And since we'll have one test, I think it will be beneficial to move `generateMouseButtonAsserts` here, so that it does not distract others.

Reply:
Done 👍